pushy/src/main/java/com/eatthepath/pushy/apns/ApnsClientHandler.java

```diff
@@ -352,17 +354,26 @@ protected void handleErrorResponse(final ChannelHandlerContext context, final in
         final HttpResponseStatus status = HttpResponseStatus.parseLine(headers.status());
 
         responseFuture.complete(new SimplePushNotificationResponse<>(responseFuture.getPushNotification(),
-                HttpResponseStatus.OK.equals(status), getApnsIdFromHeaders(headers), status.code(),
+                HttpResponseStatus.OK.equals(status), getApnsIdFromHeaders(headers),
+                getApnsUniqueIdFromHeaders(headers), status.code(),
                 errorResponse.getReason(), errorResponse.getTimestamp()));
     }
 
     private static UUID getApnsIdFromHeaders(final Http2Headers headers) {
-        final CharSequence apnsIdSequence = headers.get(APNS_ID_HEADER);
+        return getUUIDFromHeaders(headers, APNS_ID_HEADER);
+    }
+
+    private static UUID getApnsUniqueIdFromHeaders(final Http2Headers headers) {
+        return getUUIDFromHeaders(headers, APNS_UNIQUE_ID_HEADER);
+    }
+
+    private static UUID getUUIDFromHeaders(final Http2Headers headers, final AsciiString header) {
+        final CharSequence apnsIdSequence = headers.get(header);
```

Comment:
I think generalizing this method makes sense, but I think we can also go one step further and generalize this variable's name.

```suggestion
        final CharSequence uuidSequence = headers.get(header);
```